### ISSUE TITLE
resize restrictions ignore options.resize.square

### ISSUE THREAD

**Issue by Thristhart** [taye/interact.js#163]
Consider this codepen with a resizable div that does not have `square` set: http://codepen.io/Thristhart/pen/VYGKEy
No matter which way you resize it, it will not resize out of the bounds of the container.

http://codepen.io/Thristhart/pen/EaegdN
In this codepen, the div has square enabled, and now will resize out of the bounds of the container if resized from any side other than the bottom.

I wanted to make a pull request to solve this, but I was unsure where this check should be implemented.
##

**Comment by taye** [comment]
This will be a tricky one to solve. I'll look into it as soon as I can.

**Comment by gercobrandwijk** [comment]
When do you think this can be solved? 

Will that also solve the problem as described at issue https://github.com/taye/interact.js/issues/84, because I still have problems with restricting the resizing: https://jsfiddle.net/6nos0zLe/1/
After you made the blue box as wide as his parent, you are not able to make it less wide anymore. It will always go back to the full width..

**Comment by taye** [comment]
@gercobrandwijk I haven't yet tried to fix the resize issues with `square` and `preserveAspectRatio` enabled. However, #84 was fixed by #455.

```
interact('.box')
  .resizable({
    edges: { left: true, right: true, bottom: true, top: true },
    // restrict to the edges of the parent '.container' element
    restrictEdges: {
      outer: ".container",
      endOnly: true,
    },
  })
```

Here's the updated fiddle: https://jsfiddle.net/6nos0zLe/2/. You need to use [v1.3.0-alpha](https://unpkg.com/interactjs@1.3.0-alpha.5/dist/interact.js). I've written a short doc on [upgrading from v1.2](https://github.com/interactjs/docs/blob/v1.3/upgrade-v1.2-1.3.en.html.md). If you run into any difficulties, please tell me so I can improve the doc.

**Comment by taye** [comment]
#683 introduces an `aspectRatio` modifier which solves this issue by having a `modifiers` option which is a list of other modifiers to apply to the action while trying to respect the ratio.

```sh
npm install interactjs@next
```

```js
interact(target).resizable({
  edges: { left: true, bottom: true },
  modifiers: [
    interact.modifiers.aspectRatio({
       // ratio may be the string 'preserve' to maintain the starting aspect ratio,
      // or any number to force a width/height ratio
      ratio: 'preserve',
      // To add other modifiers that respect the aspect ratio,
      // put them in the aspectRatio.modifiers array
      modifiers: [interact.modifiers.restrictSize({ max: 'parent' )]
    }),
  ],
});
```
